log/klogv2/klogv2.go

```diff
@@ -14,85 +14,141 @@
 
 // Package klogv2 implements a virtual-kubelet/log.Logger using klogv2 as a backend
 //
-// You can use this by creating a klogv2 logger and calling `FromKlogv2(fields)`.
-// If you want this to be the default logger for virtual-kubelet, set `log.L` to the value returned by `FromKlogv2`
+// If you want this to be the default logger for virtual-kubelet, set `log.L` to the value returned by `New(fields)`
 //
 // We recommend reading the klog conventions to build a better understanding of levels and when they should apply
 // https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md
 package klogv2
 
 import (
+	"fmt"
+	"sort"
+	"strings"
+
 	"github.com/virtual-kubelet/virtual-kubelet/log"
 	"k8s.io/klog/v2"
 )
 
 // Ensure log.Logger is fully implemented during compile time.
 var _ log.Logger = (*adapter)(nil)
 
+type fieldMap struct {
+	log.Fields
+	processedFields string
+}
+
+func (f *fieldMap) String() string {
+	// Process if not processed before.
+	if len(f.Fields) > 0 && len(f.processedFields) == 0 {
```

Comment:
Need mutex here.

Reply:
Why? Both are immutable and only set once.

Reply:
Ah, indeed two racing first calls to this method can try and set `f.processedFields`.

Reply:
Done.